src/gitdb.ts

```diff
@@ -20,20 +38,82 @@ class GitDB {
     return new GitDB(config);
   }
 
-  public async create(collectionName: string): Promise<string> {
-    return new Promise<string>(resolve => resolve('todo: replace logic to create collection and return its name'))
+  public async readDb(): Promise<schema> {
+    const collectionDirectoryNames = await fs.readdir(this.config.dbDir);
+
+    const collectionPromises = collectionDirectoryNames.map(
+      async (collectionName) => {
+        const collectionPath = `${this.config.dbDir}/${collectionName}`;
```

Comment:
Let's extract this chunk of code into `readCollection(name)` method and account for `cache` config param